OneObjFunc.py

```diff
@@ -0,0 +1,327 @@
+# TODO -- remove these next few lines!
+# This needs to be set up in an environment
+# where snappl is available.  This will happen "soon"
+# Get Rob to fix all of this.  For now, this is a hack
+# so you can work short term.
+import sys
+import pathlib
+sys.path.insert(0, str(pathlib.Path(__file__).parent/"extern/snappl"))
+# End of lines that will go away once we do this right
+
+import numpy as np
+from astropy.io import fits
+import pandas as pd
+from roman_imsim.utils import roman_utils
+import warnings
+from astropy.utils.exceptions import AstropyWarning
+from erfa import ErfaWarning
+import scipy.sparse as sp
+from numpy.linalg import LinAlgError
+import galsim
+from AllASPFuncs import banner, fetchImages, save_lightcurve, \
+                        build_lightcurve, build_lightcurve_sim, \
+                        construct_psf_background, construct_psf_source, \
+                        makeGrid, get_galsim_SED, getWeights, generateGuess, \
+                        get_galsim_SED_list, prep_data_for_fit
+
+
+
+from simulation import simulate_images
+import yaml
+import argparse
+import os
+
+from snappl.logger import Lager
+from snappl.config import Config
+from snappl.image import OpenUniverse2024FITSImage
+
+pd.options.mode.chained_assignment = None  # default='warn'
+warnings.simplefilter('ignore', category=AstropyWarning)
+warnings.filterwarnings("ignore", category=ErfaWarning)
+
+
+def run_one_object(ID, num_total_imgs, num_detect_imgs, roman_path,
+                   sn_path, size, band, fetch_SED, use_real_images, use_roman,
+                   fit_background, turn_grid_off, adaptive_grid, npoints,
+                   make_initial_guess, initial_flux_guess, weighting, method,
+                   make_contour_grid, single_grid_point, pixel, source_phot_ops,
+                   lc_start, lc_end, do_xshift, bg_gal_flux, do_rotation, airy,
+                   mismatch_seds, deltafcn_profile, noise, check_perfection,
+                   avoid_non_linearity):
+
+    Lager.debug(f'ID: {ID}')
+    psf_matrix = []
+    sn_matrix = []
+    cutout_wcs_list = []
+    im_wcs_list = []
+    # This is a catch for when I'm doing my own simulated WCS's
```

Comment:
`im_wcs_list = []` is the catch or `util_ref = None` is the catch.

Also, `WCSs` or `set of WCS`.  No possessive apostrophe here.